diff --git a/RobotRaconteurCore/src/HardwareTransport_linux.cpp b/RobotRaconteurCore/src/HardwareTransport_linux.cpp
index 1d395c16..46f4cfb4 100644
--- a/RobotRaconteurCore/src/HardwareTransport_linux.cpp
+++ b/RobotRaconteurCore/src/HardwareTransport_linux.cpp
@@ -281,8 +281,10 @@ std::list<sockaddr_rc> BluezBluetoothConnector::GetDeviceAddresses()
             {
                 try
                 {
-                    boost::uuids::uuid svc_uuid = {0x25, 0xbb, 0x0b, 0x62, 0x86, 0x1a, 0x49, 0x74,
-                                                   0xa1, 0xb8, 0x18, 0xed, 0x54, 0x95, 0xaa, 0x07};
+                    uint8_t svc_uuid_b[] = {0x25, 0xbb, 0x0b, 0x62, 0x86, 0x1a, 0x49, 0x74,
+                                            0xa1, 0xb8, 0x18, 0xed, 0x54, 0x95, 0xaa, 0x07};
+                    boost::uuids::uuid svc_uuid;
+                    memcpy(&svc_uuid, svc_uuid_b, 16);
 
                     dbus_f->dbus_message_iter_recurse(&dict_entry, &iter_dict_val);
 
